src/keystore/InMemoryKeystore.ts

```diff
@@ -273,17 +273,16 @@ export default class InMemoryKeystore implements Keystore {
       }
       const created = nsToDate(req.createdNs)
       const recipient = toSignedPublicKeyBundle(req.recipient)
+      const myAddress = await this.getAccountAddress()
+      const theirAddress = await recipient.walletSignatureAddress()
 
       const secret = await this.v2Keys.sharedSecret(
         recipient,
         this.v2Keys.getCurrentPreKey().publicKey,
-        false
+        myAddress > theirAddress
```

Comment:
Could do something else here. Just needs to be the same irrespective of who is sending.

Reply:
To get the old deterministic tests to pass that expected a certain topic we will have to flip this to `<` but that shouldn't make difference as long as it's consistent across the SDKs